Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,261 @@
+        }
+
+        /// <summary>
+        /// Takes an array of strings and returns a jagged array with each string on ", ".
+        /// </summary>
+        public static string[][] ArrayifyBalances(string[] peopleAndBalances)
+        {
+            string[][] parsedBalances = new string[peopleAndBalances.Length][];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] tempArr = peopleAndBalances[i].Split(", ", System.StringSplitOptions.None);
+                parsedBalances[i] = tempArr;
+            }
+
+            return parsedBalances;
+        }
+
+        /// <summary>
+        /// Checks if an array is NULL or empty and returns true if any of those conditions are true.
+        /// </summary>
+        public static bool IsArrayNullOrEmpty(string[] peopleAndBalances)
+        {
+            return peopleAndBalances == null || peopleAndBalances.Length == 0;
+        }
+
+        /// <summary>
+        /// Takes an array of strings and returns that array as floats.
+        /// </summary>
+        public static float[] StringArrToFloatArr(string[] balances)
+        {
+            int failLength = 0;
+            float[] floatArr = new float[balances.Length];
+            for (int i = 0; i < floatArr.Length; i++)
+            {
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
+                else failLength++;
+            }
+            if(failLength == 0) return floatArr;
+            float[] failFloatArr = new float[floatArr.Length - failLength];
+            for (int i = 0; i < failFloatArr.Length; i++)
+            {
+                failFloatArr[i] = floatArr[i];
+            }
+            return failFloatArr;
+        }
+
+        /// <summary>
+        /// Takes an array of floats and returns a single float of the highest value.
+        /// </summary>
+        public static float HighestBalance(float[] balances)
+        {
+            float highestBalance = balances[0];
+            if (balances.Length == 0) return highestBalance;
+            for (int i = 1; i < balances.Length; i++)
+            {
+                if (balances[i] > highestBalance) highestBalance = balances[i];
+            }
+            return highestBalance;
+        }
+
+        /// <summary>
+        /// Takes an array of names and balances, then returns the highest value of all balances.
+        /// </summary>
+        /// <param name="peopleAndBalances"></param>
+        /// <returns></returns>
+        public static float HighestBalanceEver(string[] peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float highestBalanceEver = HighestBalance(StringArrToFloatArr(workingPeopleAndBalances[0][1..]));
+            if (workingPeopleAndBalances.Length == 1) return highestBalanceEver;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)